protocols/messages_test.go

```diff
@@ -103,3 +105,47 @@ func TestMessage(t *testing.T) {
 	})
 
 }
+
+func TestSortedProposals(t *testing.T) {
+	type TestCase struct {
+		Input       []consensus_channel.SignedProposal
+		Expectation []consensus_channel.SignedProposal
+	}
+	testCases := []TestCase{
+		{
+			Input: []consensus_channel.SignedProposal{
+				addProposal(types.Destination{'a'}, 1),
+				addProposal(types.Destination{'a'}, 0),
+			},
+			Expectation: []consensus_channel.SignedProposal{
+				addProposal(types.Destination{'a'}, 0),
+				addProposal(types.Destination{'a'}, 1),
+			},
+		},
+		{
+			Input: []consensus_channel.SignedProposal{
+				addProposal(types.Destination{'b'}, 1),
+				addProposal(types.Destination{'b'}, 0),
+				addProposal(types.Destination{'a'}, 1),
+				addProposal(types.Destination{'a'}, 0),
+			},
+			Expectation: []consensus_channel.SignedProposal{
+				addProposal(types.Destination{'a'}, 0),
+				addProposal(types.Destination{'a'}, 1),
+				addProposal(types.Destination{'b'}, 0),
+				addProposal(types.Destination{'b'}, 1),
+			},
+		},
+	}
```

Comment:
Include a mixture of the proposal types?

```suggestion
	testCases := []TestCase{
		{
			Input: []consensus_channel.SignedProposal{
				removeProposal(types.Destination{'a'}, 1),
				addProposal(types.Destination{'a'}, 0),
			},
			Expectation: []consensus_channel.SignedProposal{
				addProposal(types.Destination{'a'}, 0),
				removeProposal(types.Destination{'a'}, 1),
			},
		},
		{
			Input: []consensus_channel.SignedProposal{
				removeProposal(types.Destination{'b'}, 1),
				addProposal(types.Destination{'b'}, 0),
				removeProposal(types.Destination{'a'}, 1),
				addProposal(types.Destination{'a'}, 0),
			},
			Expectation: []consensus_channel.SignedProposal{
				addProposal(types.Destination{'a'}, 0),
				removeProposal(types.Destination{'a'}, 1),
				addProposal(types.Destination{'b'}, 0),
				removeProposal(types.Destination{'b'}, 1),
			},
		},
	}
```